test/integration/retryable-writes/retryable_writes.spec.test.js

```diff
@@ -2,47 +2,64 @@
 
 const { expect } = require('chai');
 const { loadSpecTests } = require('../../spec');
-const { parseRunOn } = require('../../tools/spec-runner');
+const { legacyRunOnToRunOnRequirement } = require('../../tools/spec-runner');
+const { topologySatisfies } = require('../../tools/unified-spec-runner/unified-utils');
 
 describe('Retryable Writes', function () {
   let ctx = {};
-  loadSpecTests('retryable-writes').forEach(suite => {
-    const environmentRequirementList = parseRunOn(suite.runOn);
-    environmentRequirementList.forEach(requires => {
-      const suiteName = `${suite.name} - ${requires.topology.join()}`;
-
-      describe(suiteName, {
-        metadata: { requires },
-        test: function () {
-          // Step 3: Test Teardown. Turn off failpoints, and close client
-          afterEach(function () {
-            if (!ctx.db || !ctx.client) {
-              return;
-            }
+  const retryableWrites = loadSpecTests('retryable-writes');
+
+  for (const suite of retryableWrites) {
+    describe(suite.name, function () {
+      beforeEach(async function () {
+        let utilClient;
+        if (this.configuration.isLoadBalanced) {
+          // The util client can always point at the single mongos LB frontend.
+          utilClient = this.configuration.newClient(this.configuration.singleMongosLoadBalancerUri);
+        } else {
+          utilClient = this.configuration.newClient();
+        }
 
-            return Promise.resolve()
-              .then(() =>
-                ctx.failPointName ? turnOffFailPoint(ctx.client, ctx.failPointName) : {}
-              )
-              .then(() => ctx.client.close())
-              .then(() => (ctx = {}));
-          });
+        await utilClient.connect();
 
-          suite.tests.forEach(test => {
-            it(test.description, function () {
-              // Step 1: Test Setup. Includes a lot of boilerplate stuff
-              // like creating a client, dropping and refilling data collections,
-              // and enabling failpoints
-              return executeScenarioSetup(suite, test, this.configuration, ctx).then(() =>
-                // Step 2: Run the test
-                executeScenarioTest(test, ctx)
-              );
-            });
-          });
+        const allRequirements = suite.runOn.map(r => legacyRunOnToRunOnRequirement(r));
+
+        let shouldRun = true;
+        for (const requirement of allRequirements) {
+          shouldRun =
+            shouldRun && (await topologySatisfies(this.currentTest.ctx, requirement, utilClient));
+        }
+
+        await utilClient.close();
+
+        if (!shouldRun) this.skip();
```

Comment:
After reviewing the rest of this PR, I think I answered this myself.  The runOnRequirement checker already adds a skip reason if there is one so I think adding one here isn't necessary 😄

Reply:
Right, these specs have a slightly different setup so they needed this "almost" copy of the beforeEach hook but it will still call topologySatisfies which calculates a reason.